packages/sdk-react/src/components/export/Export.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState } from "react";
+import { Modal, Box, Typography } from "@mui/material";
+import { useTurnkey } from "../../hooks/use-turnkey";
+import type { TurnkeyIframeClient } from "@turnkey/sdk-browser";
+import styles from "./Export.module.css";
+import unlockIcon from "assets/unlock.svg";
+import eyeIcon from "assets/eye.svg";
+import cautionIcon from "assets/caution.svg";
+import turnkeyIcon from "assets/turnkey.svg";
+import exportIcon from "assets/export.svg";
+
+type ExportProps = {
+  walletId: string;
+  onHandleExportSuccess: () => Promise<void>;
+  onError: (errorMessage: string) => void;
+};
+
+const Export: React.FC<ExportProps> = ({
+  walletId,
+  onHandleExportSuccess,
+  onError,
+}) => {
+  const { authIframeClient, turnkey } = useTurnkey();
+  const [exportIframeClient, setExportIframeClient] =
+    useState<TurnkeyIframeClient | null>(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isIframeVisible, setIsIframeVisible] = useState(false);
+  const TurnkeyExportIframeContainerId = "turnkey-export-iframe-container-id";
+  const TurnkeyIframeElementId = "turnkey-default-iframe-element-id";
+
+  const handleOpenModal = async () => {
+    setIsModalOpen(true);
+
+    // Wait for the modal and its content to render
+    requestAnimationFrame(async () => {
+      const iframeContainer = document.getElementById(
+        TurnkeyExportIframeContainerId
+      );
+      if (!iframeContainer) {
+        console.error("Iframe container not found.");
+        return;
+      }
+
+      const existingIframe = document.getElementById(TurnkeyIframeElementId);
+
+      if (!existingIframe) {
+        try {
+          const newExportIframeClient = await turnkey?.iframeClient({
+            iframeContainer: document.getElementById(
+              TurnkeyExportIframeContainerId
+            ),
+            iframeUrl: process.env.NEXT_PUBLIC_EXPORT_IFRAME_URL!,
+          });
+          setExportIframeClient(newExportIframeClient!);
+        } catch (error) {
+          console.error("Error initializing IframeStamper:", error);
+        }
+      }
+    });
+  };
+
+  const handleCloseModal = () => {
+    // Clear the iframe stamper
+    if (exportIframeClient) {
+      setExportIframeClient(null);
+
+      const existingIframe = document.getElementById(TurnkeyIframeElementId);
+      if (existingIframe) {
+        existingIframe.remove();
+      }
+    }
+
+    // Reset modal and iframe states
+    setIsModalOpen(false);
+    setIsIframeVisible(false);
+  };
+
+  const handleExport = async () => {
+    await exportWallet();
+    setIsIframeVisible(true);
+  };
+
+  const exportWallet = async () => {
+    try {
+      const whoami = await authIframeClient!.getWhoami();
```

Comment:
do we want to use local storage to do this check? can be pushed to future version

Reply:
https://linear.app/turnkey/issue/ENG-2018/plugnplay-use-new-sessions-api-to-replace-whoami-checks